lib/cancel-stuck-txs.js

```diff
@@ -0,0 +1,171 @@
+import pRetry from 'p-retry'
+import assert from 'node:assert'
+import * as Sentry from '@sentry/node'
+import ms from 'ms'
+import timers from 'node:timers/promises'
+
+const ROUND_LENGTH_MS = ms('20 minutes')
+
+/**
+  * @param {string} f4addr
+  * @returns 1000 oldest messages
+  */
+async function getMessagesInMempool (f4addr) {
+  const res = await pRetry(
+    () => fetch(`https://filfox.info/api/v1/message/mempool/filtered-list?address=${f4addr}&pageSize=1000`),
+    {
+      async onFailedAttempt (error) {
+        console.warn(error)
+        console.warn('Filfox request failed. Retrying...')
+      }
+    }
+  )
+  if (!res.ok) {
+    throw new Error(`Filfox request failed with ${res.status}: ${(await res.text()).trimEnd()}`)
+  }
+  /** @type {{
+     messages: {
+      cid: string;
+      from: string;
+      to: string;
+      nonce: number;
+      value: string;
+      gasLimit: number;
+      gasFeeCap: string;
+      gasPremium: string;
+      method: string;
+      methodNumber: number;
+      evmMethod: string;
+      createTimestamp: number;
+    }[];
+  }}
+  */
+  const { messages } = /** @type {any} */(await res.json())
+  return messages
+}
+
+/**
+ * @returns {Promise<{
+  "cid": string;
+  "height": number;
+  "timestamp": number;
+  "gasLimit": number;
+  "gasFeeCap": string;
+  "gasPremium": string;
+  "method": string;
+  "methodNumber": number;
+  "receipt": {
+      "exitCode": number;
+      "return": string;
+      "gasUsed": number;
+    },
+    "size": number;
+    "error": string;
+    "baseFee": string;
+    "fee": {
+      "baseFeeBurn": string;
+      "overEstimationBurn": string;
+      "minerPenalty": string;
+      "minerTip": string;
+      "refund": string;
+    },
+  }>}
+*/
+async function getRecentSendMessage () {
+  let res = await fetch('https://filfox.info/api/v1/message/list?method=Send')
+  if (!res.ok) {
+    throw new Error(`Filfox request failed with ${res.status}: ${(await res.text()).trimEnd()}`)
+  }
+  const body = /** @type {any} */(await res.json())
+  assert(body.messages.length > 0, '/message/list returned an empty list')
+  const sendMsg = body.messages.find(m => m.method === 'Send')
+  assert(!!sendMsg, 'No Send message found in the recent committed messages')
+  const cid = sendMsg.cid
+
+  res = await fetch(`https://filfox.info/api/v1/message/${cid}`)
+  if (!res.ok) {
+    throw new Error(`Filfox request failed with ${res.status}: ${(await res.text()).trimEnd()}`)
+  }
+
+  return /** @type {any} */(await res.json())
+}
+
+const cancelStuckTxs = async ({ walletDelegatedAddress, signer }) => {
+  console.log('Checking for stuck transactions...')
+
+  const messages = await getMessagesInMempool(walletDelegatedAddress)
+  const txsToCancel = messages.filter(m => {
+    return new Date(m.createTimestamp * 1000) <
+      new Date(Date.now() - (2 * ROUND_LENGTH_MS))
```

Comment:
A simpler alternative to consider:

```suggestion
    return m.createTimestamp * 1000 <
      (Date.now() - 2 * ROUND_LENGTH_MS)
```